src/main/java/net/sf/jabref/gui/contentselector/FieldContentSelector.java

```diff
@@ -161,19 +160,18 @@ private void selectionMade() {
             return;
         }
 
-        // The following is not possible at the moment since the
-        // combobox cannot be edited!
 
-        // User edited in a new word. Add it.
-        // if (comboBox.getSelectedIndex() == -1)
-        // addWord(chosen);
+        String currentText = editor.getText();
 
-        // TODO: could improve checking as not do add the same item twice
-        if (!"".equals(editor.getText())) {
+        // not the first word and no duplicate -> we need a comma
+        if (!"".equals(currentText) && !currentText.contains(chosen)) {
             editor.append(FieldContentSelector.this.delimiter);
         }
 
-        editor.append(chosen);
+        // no duplicate -> add it
+        if (!currentText.contains(chosen)) {
```

Comment:
I think it is better to reuse the KeywordList class here since keyword duplication is not as easy as it appears on first sight. For example, if the field contains "some test, something else" and I would like to add "test", then the simple "contains" solution marks this as duplicate although "same test" is not the same as "test" 😼.